interfaces/builtin/serial_port.go

```diff
@@ -50,25 +62,59 @@ func (iface *SerialPortInterface) SanitizeSlot(slot *interfaces.Slot) error {
 		panic(fmt.Sprintf("slot is not of interface %q", iface))
 	}
 
+	// We will only allow creation of this type of slot by a gadget or OS snap
+	if !(slot.Snap.Type == "gadget" || slot.Snap.Type == "os") {
+		return fmt.Errorf("serial-port slots only allowed on gadget or core snaps")
+	}
+
 	// Check slot has a path attribute identify serial device
 	path, ok := slot.Attrs["path"].(string)
 	if !ok || path == "" {
 		return fmt.Errorf("serial-port slot must have a path attribute")
 	}
 
-	// Clean the path before checking it matches the pattern
+	// Clean the path before further checks
 	path = filepath.Clean(path)
 
-	// Check the path attribute is in the allowable pattern
-	if serialAllowedPathPattern.MatchString(path) {
-		return nil
+	switch len(slot.Attrs) {
+	case 1:
```

Comment:
Note we can allow a path such as `/dev/serial-port-[a-z0-9]+` to enable custom symlinks